src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PhysicalDetailResourceTest.kt

```diff
@@ -0,0 +1,176 @@
+
+      initialiseSearchData = false
+    }
+  }
+
+  @Test
+  fun `access forbidden when no authority`() {
+    webTestClient.post().uri("/physical-detail")
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isUnauthorized
+  }
+
+  @Test
+  fun `access forbidden when no role`() {
+    webTestClient.post().uri("/physical-detail")
+      .body(
+        BodyInserters.fromValue(
+          gson.toJson(
+            PhysicalDetailRequest(
+              minHeight = 100,
+              prisonIds = listOf("LEI", "MDI")
+            )
+          )
+        )
+      )
+      .headers(setAuthorisation())
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isForbidden
+  }
+
+  @Test
+  fun `bad request when no filtering prison IDs provided`() {
+    webTestClient.post().uri("/physical-detail")
+      .body(BodyInserters.fromValue(gson.toJson(PhysicalDetailRequest(minHeight = 100, prisonIds = emptyList()))))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isBadRequest
+  }
+
+  @Test
+  fun `bad request when heights less than 0`() {
+    webTestClient.post().uri("/physical-detail")
+      .body(BodyInserters.fromValue(gson.toJson(PhysicalDetailRequest(minHeight = -100, maxHeight = -200, prisonIds = listOf("MDI")))))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isBadRequest
+  }
+
+  @Test
+  fun `bad request when heights inverted`() {
+    webTestClient.post().uri("/physical-detail")
+      .body(BodyInserters.fromValue(gson.toJson(PhysicalDetailRequest(minHeight = 100, maxHeight = 50, prisonIds = listOf("MDI")))))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isBadRequest
+  }
+
+  @Test
+  fun `can perform a detail search for ROLE_GLOBAL_SEARCH role`() {
+    webTestClient.post().uri("/physical-detail")
+      .body(BodyInserters.fromValue(gson.toJson(PhysicalDetailRequest(minHeight = 100, prisonIds = listOf("MDI")))))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isOk
+  }
+
+  @Test
+  fun `can perform a detail search for ROLE_PRISONER_SEARCH role`() {
+    webTestClient.post().uri("/physical-detail")
+      .body(BodyInserters.fromValue(gson.toJson(PhysicalDetailRequest(minHeight = 100, prisonIds = listOf("MDI")))))
+      .headers(setAuthorisation(roles = listOf("ROLE_PRISONER_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isOk
+  }
+
+  @Test
+  fun `can perform a detail search for ROLE_GLOBAL_SEARCH and ROLE_PRISONER_SEARCH role`() {
```

Comment:
Not really sure what this is testing?

Reply:
good question, it was a copy and paste from the PrisonerDetailResourceTest.  Will investigate and fix in next PR.

Reply:
Isn't testing if they have both roles it still works ? Which is OTT